### PR TITLE
CommunicatorFactory class for Internet and Lan communication

### PR DESCRIPTION
**Pull request by Prudhvis-gitrepo**
Had implemented logic, commenting has to be done and logger has to be used. So this partially fixes convict-git/KoraKaagaz#112

### CODE REVIEW

File: src/networking/CommunicatorFactory.java [hunk 1]
```diff
@@ -49,31 +71,124 @@ public synchronized static ICommunicator getCommunicator(int port){
...
+
+			/** Creating DataOutputStream that uses underlying output stream of socket. */
+			DataOutputStream out=new DataOutputStream(sock.getOutputStream());
+			
+			/** Sending message to server to get Client id of respective client */
+			out.writeUTF("GET_CLIENT_INFO");
+
+			int i=0;
+			/** reading only client id and nothing else */
+			while(i!=1 && (from=in.readUTF())!=null){
```
Comment (sravand123) [inline src/networking/CommunicatorFactory.java]: Just use in.readUTF()

File: src/networking/CommunicatorFactory.java [hunk 2]
```diff
@@ -49,31 +71,124 @@ public synchronized static ICommunicator getCommunicator(int port){
...
+			return -1;
+
+		/** This variable is used to store the client id */
+		String from="";
+		try{
+			/** Creates a stream socket and connects it to the specified port number at the specified IP address. */
+			Socket sock=new Socket(str[0],Integer.parseInt(str[1]));
+			
+			/** Creating DataInputStream that uses underlying input stream of the socket through BufferedInputStream. */
+			DataInputStream in=new DataInputStream(new BufferedInputStream(sock.getInputStream()));
```
Comment (sravand123) [inline src/networking/CommunicatorFactory.java]: remove BufferedInputStream

File: src/networking/CommunicatorFactory.java [hunk 3]
```diff
@@ -49,31 +71,124 @@ public synchronized static ICommunicator getCommunicator(int port){
...
 
-	
+
+	/**
+	* This access the xml file which consists of Server IP address and listening port at server, These details are stored 
+	* by the respective user or client.
+	*
+	* @return String[] which consists of aws server IP address and port
+	*/
+	private static String[] getServerInfo(){
```
Comment (sravand123) [inline src/networking/CommunicatorFactory.java]: change from private to default

File: src/networking/CommunicatorFactory.java [hunk 4]
```diff
@@ -95,40 +212,54 @@ public static ClientInfo getClientInfo(){
...
+
+					/** Initially checks for internet availability and server(aws or any specified IP and port in xml file) availability */ 
+					port =getClientId();
 
+					/** 
+					* Initialising the CommunicatorFactory class to create Lan communicator as there is 
+					* no internet or aws availability 
+					*/ 
+					if(port==-1){
+						setTypeOfCommunicator("LAN");
```
Comment (sravand123) [inline src/networking/CommunicatorFactory.java:230]:
```port =sock,getLocalPort()
return new ClientInfo(ip,port)```

**Review by sravand123 — CHANGES_REQUESTED** [review]

File: src/networking/CommunicatorFactory.java (hunk 3 above)
Comment (Prudhvis-gitrepo) [reply src/networking/CommunicatorFactory.java]: This was not there in the discussed design. I have written this function for my convenience. So Is this new extra requirement to be done??

Comment (sravand123) [reply src/networking/CommunicatorFactory.java]: Yeah

**Pull request merged**